File: LeetCode题目集/226. Invert Binary Tree(easy)-迭代法.java
import java.util.LinkedList;
import java.util.Queue;

/*
 * @Author: SourDumplings
 * @Date: 2019-08-23 21:46:48
 * @Link: https://github.com/SourDumplings/
 * @Email: changzheng300@foxmail.com
 * @Description: https://leetcode.com/problems/invert-binary-tree/
 * 
 * 层序遍历交换各结点的左右子树即可
 */

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution
{
    public TreeNode invertTree(TreeNode root)
    {
        if (root == null)
        {
            return null;
        }
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        while (!q.isEmpty())
        {
            TreeNode r = q.remove();
            TreeNode temp = r.left;
            r.left = r.right;
            r.right = temp;
            if (r.left != null)
                q.add(r.left);
            if (r.right != null)
                q.add(r.right);
        }
        return root;
    }
}

File: OJ practices/LeetCode题目集/226. Invert Binary Tree(easy)-迭代法.cpp
/**
 * @file 226. Invert Binary Tree(easy)-迭代法.cpp
 * @author SourDumplings (sourdumplings@qq.com)
 * @brief
 * https://leetcode.cn/problems/invert-binary-tree/?envType=study-plan-v2&envId=top-interview-150
 * @version 1.0.0
 * @date 2024-12-09
 *
 * @copyright Copyright (c) 2024 SourDumplings
 *
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
	TreeNode *invertTree(TreeNode *root)
	{
		if (root == nullptr) { return nullptr; }

		queue<TreeNode *> q;
		q.push(root);
		while (!q.empty())
		{
			TreeNode *node = q.front();
			q.pop();
			swap(node->left, node->right);
			if (node->left) { q.push(node->left); }
			if (node->right) { q.push(node->right); }
		}
		return root;
	}
};